utils/core/src/serde/byte_writer.rs

```diff
@@ -100,7 +100,11 @@ pub trait ByteWriter: Sized {
     /// Serializes all `elements` and writes the resulting bytes into `self`.
     ///
     /// This method does not write any metadata (e.g. number of serialized elements) into `self`.
-    fn write_many<S: Serializable>(&mut self, elements: &[S]) {
+    fn write_many<S, T>(&mut self, elements: T)
+    where
+        T: IntoIterator<Item = S>,
+        S: Serializable,
+    {
```

Comment:
Question: is there a reason not to define this as:
```Rust
fn write_many<S, T>(&mut self, elements: T)
where
    T: Iterator<Item = S>,
    S: Serializable,
{
```

Reply:
`IntoIterator` covers more use cases, and basically remove the need for the user to do `write_many(vec.iter())`, a `write_many(vec)` will suffice. At the same time `write_many(vec.iter())` is still supported, since an [`Iterator` implements this trait](https://doc.rust-lang.org/core/iter/trait.IntoIterator.html#impl-IntoIterator-for-I)

Reply:
Ah - so, if we didn't want to pass the ownership of the object to `write_many()`, we could still write `write_many(vec.iter())`, right?

Reply:
`vec.iter()` works, also just `&vec`. Here example:

```rust
use std::fmt::Debug;

fn iter<T, I>(vs: T)
where
    T: IntoIterator<Item = I>,
    I: Debug,
{
    for v in vs {
        print!("{:?} ", v);
    }
    print!("\n");
}

fn main() {
    let vs = vec![1, 2, 3, 4];

    iter(&vs);
    iter(&vs);
    iter(vs.iter());
    iter(vs.iter());
}
```

Will print:

```
1 2 3 4
1 2 3 4
1 2 3 4
```

Reply:
One important note, `IntoIterator` is a trait that specifies how to convert a type to an `Iterator`. And that can be owning or not. It is not like `into_iter` that by convention is the owning version of the iterator.